openvdb/openvdb/tools/Merge.h

```diff
@@ -966,12 +977,35 @@ bool CsgUnionOrIntersectionOp<TreeT, Union>::operator()(LeafT& leaf, size_t) con
             continue;
         }
 
-        for (Index i = 0 ; i < LeafT::SIZE; i++) {
-            const ValueT& newValue = mergeLeaf->getValue(i);
-            const bool doMerge = Union ? newValue < leaf.getValue(i) : newValue > leaf.getValue(i);
-            if (doMerge) {
-                leaf.setValueOnly(i, newValue);
-                leaf.setActiveState(i, mergeLeaf->isValueOn(i));
+        if (mPruneCancelledTiles) {
+            bool allnegequal = true;
+            for (Index i = 0 ; i < LeafT::SIZE; i++) {
+                const ValueT& newValue = mergeLeaf->getValue(i);
+                const ValueT& oldValue = leaf.getValue(i);
+                allnegequal &= oldValue == math::negative(newValue);
+                const bool doMerge = Union ? newValue < oldValue : newValue > oldValue;
+                if (doMerge) {
+                    leaf.setValueOnly(i, newValue);
+                    leaf.setActiveState(i, mergeLeaf->isValueOn(i));
+                }
+            }
+            if (allnegequal) {
+                // If two diffed tiles have the same values of opposite signs,
+                // we know they have both the same distances and gradients.
+                // Thus they will cancel out.
+                if (Union) { leaf.fill(math::negative(this->background()), false); }
+                else { leaf.fill(this->background(), false); }
```

Comment:
I don't like this->background(), but this was done at the start of this method!
const ValueT background = Union ? this->background() : -this->background();

Reply:
I don't understand, why can't you just do:
```
leaf.fill(background, false);
```
?

Reply:
That background is opposite sign to what we want.   So it could be
leaf.fill(math::negative(background), false);
but that would be rather confusing... I like the background always meaning positive background.

Reply:
Ah yes, makes sense.